### ISSUE TITLE
Combined report giving wrong answers

### ISSUE THREAD

**Issue by ghost** [coveragepy/coveragepy#939]
**Scenario:**
I'm running all my tests using python 3.8 and unittest. I've divided tests in "unit" and "integration" tests, where unittests start in "test_" and are run on one single docker container and integration tests start in "itest_" and are run from a "home" container using a network of docker containers. Some files have only unit tests, some files have only integration tests, some files have both. The test files themselves are excluded from the coverage report. After testing, I copy the .coverage file onto my host machine, naming them .coverage.unit and .coverage.integration respectively. Each test, before it runs, checks for a pre-existing coverage file of the other test. If it exists, it copies it onto the container and uses `coverage combine` before generating the report using `coverage report -m`.

**Problem:**
If I run my unit tests first, the result is this:

```
Name                              Stmts   Miss  Cover   Missing
---------------------------------------------------------------
core/A.py                            61      0   100%
core/B.py                           159    121    24%   46-63, 68-69, 72, 82, 97-114, 119-121, 124-125, 135, 139-188, 193-205, 208-212, 215-221, 226-258, 261-272, 277, 282
core/C.py                            54      0   100%
core/D.py                             7      0   100%
core/E.py                           203     10    95%   84-87, 154-155, 242-243, 251-252
core/F.py                            11      0   100%
core/G.py                            14      0   100%
core/H.py                            11      0   100%
core/I.py                            13      0   100%
msg/A.py                             62      0   100%
---------------------------------------------------------------
TOTAL                               595    131    78%
```

However, if I now run the integration tests, combining their result with the result of the unit tests, I get the exact same result.

Now If I reverse this and run the integration tests first (without any previous result files present), I get the following:

```
Name                                            Stmts   Miss  Cover   Missing
-----------------------------------------------------------------------------
core/A.py                                          61     12    80%   52, 74-75, 83-84, 89, 95, 98, 101, 104, 113-114
core/B.py                                         159     37    77%   46-63, 68-69, 72, 82, 108, 111, 156-161, 163-168, 185, 196-198, 201, 210, 217, 252-255, 282
core/C.py                                          54     17    69%   30, 44, 54-55, 60, 73-76, 94-100, 104, 107, 110, 113
core/D.py                                           7      0   100%
core/E.py                                         203     69    66%   47, 84-87, 102, 104, 115, 139-146, 153-163, 171-174, 182, 217-256, 264-290, 318-320, 331, 337, 343, 349, 367, 374, 380, 386
core/F.py                                          11      0   100%
core/G.py                                          14      0   100%
core/H.py                                          11      0   100%
core/I.py                                          13      0   100%
msg/X.py                                           64      1    98%   127
msg/Y.py                                           28      0   100%
msg/Z.py                                          151     22    85%   91, 98-102, 153, 158-162, 220-225, 260-264, 303-307
msg/A.py                                           62      6    90%   79, 95-96, 130-132
-----------------------------------------------------------------------------
TOTAL                                             838    164    80%
```

When I then run the unit tests to get the combined result, the table stays mainly the same with the exception of A.py, for which it now looks like this:
```
core/A.py                                          63     27    57%   52, 63, 69-87, 93-107, 112-117, 123-140
```

What I expected:
- Integration tests test more files than unit tests
- coverage of integration tests alone (without unit tests) would be lower for some classes, e.g. A.py
- 100% coverage for A.py after running both sets of tests as `max(1, 63/100)==1`

What I didn't expect:
- different results for combine depending of the order in which tests are run - surely it should "merge" all results to find the overall coverage.
- A.py has mysteriously gained two extra statements
- cover for A.py after running integration and then unit tests has gone down significantly

**TL;DR:**
I think there's something fishy going on in the combine method.

**Comment by nedbat** [comment]
Hi, sorry I let this sit for so long.  Without more details, it might be hard to debug.

A common problem with combining data is having the "same" source code at different paths for different runs.  With your different Docker containers, this sounds like this could be happening to you.  The `[paths]` section of the configuration file is the tool for mapping different paths to the same source file: https://coverage.readthedocs.io/en/coverage-5.5/config.html#config-paths  Maybe that will solve the problem for you?

**Comment by nedbat** [comment]
Feel free to re-open this if you come back to it.
